plugins/gcs/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 3.19...3.24 FATAL_ERROR)
+
+project(
+  gcs
+  DESCRIPTION "GCS plugin for Tenzir"
+  LANGUAGES CXX)
+
+include(CTest)
+
+find_package(Tenzir REQUIRED)
+
+TenzirRegisterPlugin(
+  TARGET gcs
+  ENTRYPOINT src/plugin.cpp)
```

Comment:
Afaict this adds no additional dependencies. Should this be a builtin instead of a plugin to keep the build process lean?

(The same applies to S3.)

Reply:
Sure, but then the Arrow CMake sub-dependencies `ARROW_FILESYSTEM`, `ARROW_S3` and `ARROW_GCS` become mandatory then.
OK?

Reply:
I'd say yes. @tobim are they available in Nix?

Reply:
GCS also adds the `google-cloud-sdk` dependency.

Reply:
It is `google-cloud-cpp`, and that needs to be patched to build correctly in static only-mode.

Reply:
I got `google-cloud-cpp` to build and install correctly in static-only mode. Is it OK if I push to this branch?

Reply:
YES!